lib/ansible/modules/crypto/openssl_certificate.py

```diff
@@ -0,0 +1,788 @@
+            raise CertificateError(
+                'The private key %s does not exist' % self.privatekey_path
+            )
+
+        if not os.path.exists(self.csr_path):
+            raise CertificateError(
+                'The certificate signing request file %s does not exist' % self.csr_path
+            )
+
+        if not os.path.exists(self.accountkey_path):
+            raise CertificateError(
+                'The account key %s does not exist' % self.accountkey_path
+            )
+
+        if not os.path.exists(self.challenge_path):
+            raise CertificateError(
+                'The challenge path %s does not exist' % self.challenge_path
+            )
+
+        if not self.check(module, perms_required=False) or self.force:
+            try:
+                p = subprocess.Popen([
+                    'acme-tiny',
+                    '--account-key', self.accountkey_path,
+                    '--csr', self.csr_path,
+                    '--acme-dir', self.challenge_path], stdout=subprocess.PIPE)
+                crt = p.communicate()[0]
+                with open(self.path, 'wb') as certfile:
+                    certfile.write(str(crt))
+            except OSError as exc:
+                raise CertificateError(exc)
+
+        file_args = module.load_file_common_arguments(module.params)
+        if module.set_fs_attributes_if_different(file_args, False):
+            self.changed = True
+
+    def dump(self):
+
+        result = {
+            'changed': self.changed,
+            'filename': self.path,
+            'privatekey': self.privatekey_path,
+            'accountkey': self.accountkey_path,
+            'csr': self.csr_path,
+        }
+
+        return result
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            state=dict(default='present', choices=['present', 'absent'], type='str'),
+            path=dict(required=True, type='path'),
+            provider=dict(choices=['selfsigned', 'assertonly', 'acme'], type='str'),
+            force=dict(default=False, type='bool'),
+            csr_path=dict(type='path'),
+
+            # General properties of a certificate
+            privatekey_path=dict(type='path'),
+            privatekey_passphrase=dict(type='path', no_log=True),
+            signature_algorithms=dict(type='list'),
+            subject=dict(type='dict'),
+            issuer=dict(type='dict'),
+            has_expired=dict(default=False, type='bool'),
+            version=dict(type='int'),
+            keyUsage=dict(type='list'),
+            keyUsage_strict=dict(default=False, type='bool'),
+            extendedKeyUsage=dict(aliases=['extKeyUsage'], type='list'),
+            extendedKeyUsage_strict=dict(aliases=['extKeyUsage_strict'], default=False, type='bool'),
+            subjectAltName=dict(type='list'),
+            subjectAltName_strict=dict(default=False, type='bool'),
+            notBefore=dict(type='str'),
+            notAfter=dict(type='str'),
```

Comment:
if you change it to snake, it would be possible to add an alias like `notAfter=dict(type='str', aliases=['not_after']),` and in the docs you can change it the other way around (to not to touch everything):

~~~yaml
    not_after:
        required: false
        aliases: [ notAfter ]
        description:
            - The certificate must expire at this point in time. The timestamp is formatted as an ASN.1 TIME.
~~~